phys2denoise/metrics/utils.py

```diff
@@ -332,3 +335,97 @@ def export_metric(
                 )
 
     return fileprefix
+
+
+def return_physio_or_metric(*, return_physio=True):
+    """
+    Decorator to check if the input is a Physio object.
+
+    Parameters
+    ----------
+    func : function
+        Function to be decorated
+
+    Returns
+    -------
+    function
+        Decorated function
+    """
+
+    def determine_return_type(func):
+        metrics_with_lags = ["respiratory_variance_time"]
```

Comment:
We did not implemented it yet, but technically all metrics could have lags - and all metrics could be convolved. Is this a way to check the type of output you get?

Reply:
Could you instead make a check on the provided arguments to the function, to check whether convolution and/or lag happened?

Reply:
I'll change to detect lagged metrics, since it's dependent on the argument. However convolution is now only done in some metrics without being specified in the arguments and can't be detected easily if you don't read the code. Also you can't deduce that the metric is convolved from the dimensions of the metric output.

I can open an issue about it if you want